File: 70. 爬楼梯/climbStairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function (n) {
  let arr = new Array(n + 1);
  arr[0] = arr[1] = 1;
  for (let i = 1; i < n; i++) {
    arr[i + 1] = arr[i] + arr[i - 1];
  }
  return arr[n];
};

File: 70. 爬楼梯/climbStairs.cpp
#include "../stdc++.h"
using namespace std;

class Solution {
public:
    // 动态规划
    // 第 n 层的种数 = 第 n - 1 层的种数 + 第 n - 2 层的种数
    int climbStairs(int n) {
        int p = 0, q = 0, r = 1;
        for (int i = 1; i <= n; ++i) {
            p = q; 
            q = r; 
            r = p + q;
        }
        return r;
    }
};